### PR TITLE
docs: write API document

### PR DESCRIPTION
**Pull request by moyuerrr**

### CODE REVIEW

**Review by RuipuCui — APPROVED** [review]
good job! Thank you for your effort

File: API document.md [hunk 1]
```diff
@@ -0,0 +1,832 @@
...
+
+- request path: /login
+
+- request method: PUT
+
+- request arguments type: json
+
+  - name: must
+
+  - password: must
```
Comment (Leshangpang) [inline API document.md:66]: Is there any requirements on the password? i.e. min length, capital letters or numbers needed etc. If so, do you think this should be displayed on the sign up page?

**Review by Leshangpang — APPROVED** [review]
I think it looks good. I just have a question about requirements for the password (see comment)

File: API document.md (hunk 1 above)
Comment (moyuerrr) [reply API document.md:66]: I think the password should be at least 6 characters. Other requirements are unnecessary. The dynamic detection of password format is needed to be realized on sign up page by front-end.

Comment (Leshangpang) [reply API document.md:66]: @RuipuCui Can you pls look into this further since you are responsible for the sign up page?

**Pull request merged**